### PR TITLE
Fix render html widget

### PR DESCRIPTION
**Pull request by nleush**
Fix rendering issues, mentioned in #42:
 * allow render html widget, not only card with image
 * allow render html widget with Iframely `embed.js` script, whitelisted by regular expression

@barisusakli is it safe to change `sanitizeConfig` with hook and allow generic `script` tag? This is required by Iframely widgets for resizing and other functionality. We use it only with Iframely `embed.js` script. Plugin will not render other scripts but image card will be rendered as fallback for that cases.

We hope `sanitizeConfig` works only in scope of one plugin and this will not violate overall security.

### CODE REVIEW

**Comment by barisusakli** [comment]
as far as I know modified sanitizeConfig will effect all plugin and core, @julianlam can confirm.

File: library.js [hunk 1]
```diff
@@ -153,11 +160,18 @@ iframely.replace = function(raw, options, callback) {
 
 					var icon = getIcon(embed);
 					var image = getImage(embed);
-					if (image) {
+					var scriptSrc = getScriptSrc(embedHtml);
+					// Allow only `iframe.ly/embed.js` script.
+					var isIframelyWidget = scriptSrc && /iframe\.ly\/embed\.js$/.test(scriptSrc);
+					var isSanitized = !scriptSrc || isIframelyWidget;
```
Comment (iparamonau) [inline library.js:164]: Shouldn't the name here be `isNotSanitized` for sanity?

Comment (nleush) [reply library.js:164]: `sanitized` means safe for me here.

**Comment by julianlam** [comment]
Yes, unfortunately @barisusakli is correct, updating the sanitizeConfig will update it globally, and so allowing `script` will in effect be fairly dangerous **but only if there are no parsers installed**.

For example, iframely allowing `script` will mean a plugin can inject script tags into the post body, but if the forum still uses markdown, then markdown will still strip out the script tags.

**Comment by iparamonau** [comment]
So what's the way to whitelist our script? We could probably add it once in the `<head>` of the document, or better yet - source if from GitHub/NPM and load of the same NodeBB server:

https://github.com/itteco/embedjs

**Comment by barisusakli** [comment]
I feel like adding it to the body with `filter:scripts.get` should work.

**Comment by nleush** [comment]
> I feel like adding it to the body with `filter:scripts.get` should work.

Thanks @barisusakli , it works! Didn't found it on autogenerated docs. 

I still have one more thing to do with this task and looking for your help.

We need to call command `iframely.load();` after all widgets loaded on page. This includes dynamic preview case in editor. This command actually renders widgets and starts listening important UI events from it.

Can you suggest something to do that?

**Comment by barisusakli** [comment]
You should be able to call `iframely.load` in the `action:ajaxify.end` hook client side.
That hook gets called after every page change (after everything gets rendered).

For the composer, when the preview window content changes we fire the hook `action:composer.preview` so you an also call `iframely.load` in there.

**Comment by barisusakli** [comment]
You might want to add `action:posts.loaded` as well ,this gets triggered when you are on a topic page and a new post is sent to the client. If that post has iframely content  I think you needto called iframely.load as well. Keep in mind this hook can get called many times the new posts  are passed to this hook in the posts fields.

**Comment by nleush** [comment]
@barisusakli thanks, all events work for me.

**Comment by nleush** [comment]
@barisusakli could you please publish new plugin version, as this changes are critical?

**Comment by barisusakli** [comment]
0.7.13 published, I will update on our community as well.

**Pull request merged**